## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 488538 -p kmscon -p nixosTests.kmscon`
Commit: [`c7df04ed4a23afb175739d5ff0dab221acf77662`](https://github.com/NixOS/nixpkgs/commit/c7df04ed4a23afb175739d5ff0dab221acf77662) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/c7df04ed4a23afb175739d5ff0dab221acf77662..pull/488538/head))
Merge: [`4d24ddcc8557eadfe328bb3927a00ed18ad363f1`](https://github.com/NixOS/nixpkgs/commit/4d24ddcc8557eadfe328bb3927a00ed18ad363f1)

Logs: https://github.com/ccicnce113424/nixpkgs-review-gha/actions/runs/21809112832


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>nixosTests.kmscon</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>kmscon</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>nixosTests.kmscon</summary>
<pre>machine: sending keys &#x27;foobar\n&#x27;
machine: (finished: sending keys &#x27;foobar\n&#x27;, in 0.07 seconds)
machine: waiting for alice@machine to appear on screen
machine # [  919.685406] systemd[1]: Starting Cleanup of Temporary Directories...
machine # [  919.847210] systemd[1]: systemd-tmpfiles-clean.service: Deactivated successfully.
machine # [  919.850890] systemd[1]: Finished Cleanup of Temporary Directories.
machine: Last OCR attempt failed. Text was: [&#x27;machine login\n\x0c&#x27;, &#x27;machine login: |\n\x0c&#x27;, &#x27;machine login: |\n\x0c&#x27;]
!!! Test &quot;ensure we can open a tty&quot; failed with error: &quot;action timed out after 908.59 seconds (timeout=900)&quot;
!!! Traceback (most recent call last):
!!!   File &quot;&lt;string&gt;&quot;, line 9, in &lt;module&gt;
!!!     machine.wait_for_text(&quot;alice@machine&quot;)
!!! 
!!! RequestedAssertionFailed: action timed out after 908.59 seconds (timeout=900)
cleanup
kill machine (pid 9)
qemu-system-x86_64: terminating on signal 15 from pid 6 (/nix/store/slhpx9glq7vl99bwi93bgrhn3syv98s1-python3-3.13.11/bin/python3.13)
vde_switch: EOF data port: Interrupted system call
kill vlan (pid 7)
(finished: cleanup, in 0.01 seconds)
vde_switch: Caught signal 15, cleaning up and exiting</pre>
</details>
</details>

--- a/nixos/tests/kmscon.nix
+++ b/nixos/tests/kmscon.nix
@@ -29,8 +29,7 @@
   enableOCR = true;
 
   testScript = ''
-    machine.succeed(":")
-    # ^ this create a screen
+    machine.wait_for_unit("multi-user.target")
 
     with subtest("ensure we can open a tty"):
       machine.wait_for_text("machine login:")
@@ -38,6 +37,8 @@
       machine.wait_for_text("Password:")
       machine.send_chars("foobar\n")
       machine.wait_for_text("alice@machine")
+      machine.send_chars("echo $TERM\n")
+      machine.wait_for_text("xterm-256color")
       machine.screenshot("tty.png")
   '';
 }

--- a/pkgs/by-name/km/kmscon/package.nix
+++ b/pkgs/by-name/km/kmscon/package.nix
@@ -12,6 +12,7 @@
   pango,
   pkg-config,
   docbook_xsl,
+  docbook_xml_dtd_42,
   libxslt,
   libgbm,
   ninja,
@@ -60,9 +61,22 @@ stdenv.mkDerivation (finalAttrs: {
     libxslt # xsltproc
   ];
 
+  outputs = [
+    "out"
+    "man"
+  ];
+
   patches = [
     ./sandbox.patch # Generate system units where they should be (nix store) instead of /etc/systemd/system
   ];
+
+  postPatch = ''
+    for i in ./docs/man/*.in; do
+      substituteInPlace "''${i}" \
+        --replace-fail "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd" \
+                       "${docbook_xml_dtd_42}/xml/dtd/docbook/docbookx.dtd"
+    done
+  '';
 
   passthru = {
     tests.kmscon = nixosTests.kmscon;

